application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/router/FilterTransitWhenDirectModeIsEmpty.java

```diff
@@ -53,21 +55,35 @@
 public class FilterTransitWhenDirectModeIsEmpty {
 
   private final StreetMode originalDirectMode;
+  private final boolean pageCursorExists;
 
-  public FilterTransitWhenDirectModeIsEmpty(RequestModes modes) {
-    this(modes.directMode);
+  public FilterTransitWhenDirectModeIsEmpty(RequestModes modes, boolean pageCursorExists) {
+    this(modes.directMode, pageCursorExists);
   }
 
-  public FilterTransitWhenDirectModeIsEmpty(StreetMode originalDirectMode) {
+  public FilterTransitWhenDirectModeIsEmpty(
+    StreetMode originalDirectMode,
+    boolean pageCursorExists
+  ) {
     this.originalDirectMode = originalDirectMode;
+    this.pageCursorExists = pageCursorExists;
   }
 
+  /**
+   * If the direct mode was not set and if the page cursor does not exist,
+   * the WALK mode is returned for filtering itineraries.
+   */
```

Comment:
This class feels really weird in many ways. The name makes it sound like this is a filter but this actually isn't one. This class has too many responsibilities in my opinion and this pr makes it even harder to follow the logic. It feels a bit hacky how we need to set the direct mode back-and-forth in many places. However, we have plans in the future to perhaps refactor the "direct filter" logic a bit so maybe this isn't the right time to do refactoring here.

Reply:
We decided this was out of scope for this PR in the dev meeting, but would be nice to change